pkg/util/fsnotification/fsnotification.go

```diff
@@ -0,0 +1,52 @@
+package fsnotification
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// AddRecursiveWatch handles adding watches recursively for the path provided
+// and its subdirectories.
```

Comment:
Should document here that `path` must be a directory, and if it is not, we silently ignore it.